src/Streamly/Internal/Data/Array.hs

```diff
@@ -239,3 +241,14 @@ streamFold f arr = f (toStream arr)
 {-# INLINE getIndexUnsafe #-}
 getIndexUnsafe :: Array a -> Int -> a
 getIndexUnsafe = indexArray
+
+-- | Update multiple indices of an array.
+overlay :: (PrimMonad m) => Array a -> SerialT m (Int, a) -> m (Array a)
+overlay arr m = do
```

Comment:
Instead of `overlay`, we can use the same API for this purpose as we are using in `src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs` i.e. `putIndices`.

Can you implement this on the lines `putIndices` with the same signature?